src/mds/MDSRank.cc

```diff
@@ -610,7 +610,7 @@ MDSRank::~MDSRank()
 void MDSRankDispatcher::init()
 {
   objecter->init();
-  messenger->add_dispatcher_head(objecter);
+  messenger->add_dispatcher_head(objecter, 1000);
```

Comment:
We want the objecter ahead of the MDSDaemon dispatcher. I'll change it to avoid a magic number.

Reply:
You can still rely on the `_head` suffix to put it ahead of any other handlers at the same priority, can't you? Maybe it'd mean that the other place would need to use `_tail(high_priority)`

Reply:
I would prefer to avoid relying on `_head` or `_tail` when we are already using the priorities.

Reply:
head and tail have forever been the way how messenger manages the dispatchers, so it's not going to be a surprise to anyone. Also, I think it's a neat trick to reduce the number of the named priority levels because it creates x2 effective priorities.

I also understand your desire to unify the interface, but in that case, we should eliminate the ambiguity of _head and _tail and rely on the priorities alone to do the ordering for us. Then I'd expect something like

```
add_dispatcher(&dispatcher, ::priority_high - 1); // before all high priority handlers
```